elide-integration-tests/src/test/java/com/yahoo/elide/async/integration/tests/TableExportIT.java

```diff
@@ -0,0 +1,1441 @@
+
+        String baseUrl = "/";
+        // Principal is Owner
+        response = elide.get(baseUrl, "/tableExport/" + id, new MultivaluedHashMap<>(), ownerUser, NO_VERSION);
+        assertEquals(HttpStatus.SC_OK, response.getResponseCode());
+
+        // Principal has Admin Role
+        response = elide.get(baseUrl, "/tableExport/" + id, new MultivaluedHashMap<>(), securityContextAdminUser, NO_VERSION);
+        assertEquals(HttpStatus.SC_OK, response.getResponseCode());
+
+        // Principal without Admin Role
+        response = elide.get(baseUrl, "/tableExport/" + id, new MultivaluedHashMap<>(), securityContextNonAdminUser, NO_VERSION);
+        assertEquals(HttpStatus.SC_NOT_FOUND, response.getResponseCode());
+    }
+
+    /**
+     * Various tests for a JSONAPI query as a TableExport Request with asyncAfterSeconds value set to 7.
+     * asyncAfterSeconds is more than 10.
+     * @throws InterruptedException
+     */
+    @Test
+    public void asyncAfterBeyondMax() throws InterruptedException {
+        String expected = "{\"errors\":[{\"detail\":\"Invalid value: Invalid Async After Seconds\"}]}";
+        AsyncDelayStoreTransaction.sleep = true;
+
+        //Create Async Request
+        given()
+                .contentType(JSONAPI_CONTENT_TYPE)
+                .body(
+                        data(
+                                resource(
+                                        type("tableExport"),
+                                        id("edc4a871-dff2-4054-804e-e80075cf831f"),
+                                        attributes(
+                                                attr("query", "/book?sort=genre&fields%5Bbook%5D=title"),
+                                                attr("queryType", "JSONAPI_V1_0"),
+                                                attr("status", "QUEUED"),
+                                                attr("asyncAfterSeconds", "70"),
+                                                attr("resultType", "CSV")
+                                        )
+                                )
+                        ).toJSON())
+                .when()
+                .post("/tableExport")
+                .then()
+                .statusCode(org.apache.http.HttpStatus.SC_BAD_REQUEST)
+                .body(equalTo(expected));
+
+    }
+
+    /**
+     * Reset sleep delay flag after each test.
+     */
+    @AfterEach
+    public void sleepDelayReset() {
+
+        AsyncDelayStoreTransaction.sleep = false;
+    }
+
+    private JsonNode toJsonNode(String query, Map<String, Object> variables) {
+        ObjectNode graphqlNode = JsonNodeFactory.instance.objectNode();
+        graphqlNode.put("query", query);
+        if (variables != null) {
+            graphqlNode.set("variables", mapper.valueToTree(variables));
+        }
+        return graphqlNode;
+    }
+
+    private String getStoredFileContents(String id) throws IOException {
+        StringBuilder lines = new StringBuilder();
```

Comment:
What if we leveraged the result storage engine here?  Then this code doesn't need to know the location of the temp file (see earlier comment).

Reply:
Scratch that.  I think we can remove this function (see earlier comment).